homeassistant/components/nuki/config_flow.py

```diff
@@ -99,14 +127,26 @@ async def async_step_validate(self, user_input=None):
                 errors["base"] = "unknown"
 
             if "base" not in errors:
-                await self.async_set_unique_id(info["ids"]["hardwareId"])
+                existing_entry = await self.async_set_unique_id(
+                    info["ids"]["hardwareId"]
+                )
+                if self._reauth and existing_entry:
+                    self.hass.config_entries.async_update_entry(
+                        existing_entry, data=conf
+                    )
+                    await self.hass.config_entries.async_reload(existing_entry.entry_id)
+                    return self.async_abort(reason="reauth_successful")
                 self._abort_if_unique_id_configured()
                 return self.async_create_entry(
-                    title=info["ids"]["hardwareId"], data=user_input
+                    title=info["ids"]["hardwareId"], data=conf
                 )
 
-        data_schema = self.discovery_schema or USER_SCHEMA
+        if self._reauth:
+            return self.async_show_form(
```

Comment:
We shouldn't move directly to another step's form and not share errors between different steps. The errors are specific to the form where they came from.

The validation step should be made a helper instead of a step method and it can raise errors instead that we catch in the calling steps.

Reply:
I changed it now to also use the `validate()` function directly. Did you mean it like this, or should the `errors["base"]` part put in a common function too?